buildSrc/src/main/java/com/uber/okbuck/core/model/base/Scope.java

```diff
@@ -212,13 +218,31 @@ public static Scope from(Project project, Configuration configuration) {
     }
 
     public Set<String> getAnnotationProcessors() {
+        Set<ExternalDependency.VersionlessDependency> firstLevelDependencies = configuration
```

Comment:
I didn't wanted to do it as part of that because we only need this for annotation processor configurations. 
I think keeping `configuration.getAllDependencies()` as a field should work?

Reply:
Actually we seem to be only using scope during task execution time, so this might be ok. We can re-eval later when we refactor for removing legacy ap support